zha/application/platforms/fan/__init__.py

```diff
@@ -421,20 +413,54 @@ def __init__(
     @functools.cached_property
     def preset_modes_to_name(self) -> dict[int, str]:
         """Return a dict from preset mode to name."""
-        return IKEA_PRESET_MODES_TO_NAME
+        return {1: PRESET_MODE_AUTO}
 
     @functools.cached_property
     def speed_range(self) -> tuple[int, int]:
         """Return the range of speeds the fan supports. Off is not included."""
-        return IKEA_SPEED_RANGE
+
+        # 1 is not a speed, but auto mode and is filtered out in async_set_percentage
+        return 1, 10
 
     @property
-    def default_on_percentage(self) -> int:
-        """Return the default on percentage."""
-        return int(
-            (100 / self.speed_count) * self.preset_name_to_mode[PRESET_MODE_AUTO]
+    def percentage(self) -> int | None:
+        """Return the current speed percentage."""
+        self.debug(f"fan_speed is {self._fan_cluster_handler.fan_speed}")
```

Comment:
I think we can remove this debug line. Seems a bit spammy, as this seems to be called often, for some reason. (Not related to this PR though)
We also already get multiple debug lines elsewhere when the `fan_speed` attribute changes.
```suggestion
```